src/game/SpellEffects.cpp

```diff
@@ -5457,6 +5457,33 @@ void Spell::EffectScriptEffect(SpellEffectIndex eff_idx)
                     unitTarget->RemoveAurasAtMechanicImmunity(IMMUNE_TO_ROOT_AND_SNARE_MASK, 30918, true);
                     break;
                 }
+                
+                
+                case 32027:				// Expedition Flare
+			{
+				if (!unitTarget)
+					return;
+
+				uint32 creature_id;
+				uint32 rand = urand(0, 99);
+
+				//summon Expedition Preserver or Expedition Scout 
+				if (rand < 50)
+					creature_id = 18194;
+				else 
+					creature_id = 18126;
+		
```

Comment:
You can replace the whole if/else shabang as well as the 2 variables if you copy the method used for this dummy effect: https://github.com/cmangos/mangos-tbc/blob/master/src/game/SpellEffects.cpp#L799 (which by the way is superb code style as well)